apps/src/blockly/eventHandlers.js

```diff
@@ -0,0 +1,42 @@
+// A custom version of Blockly's Events.disableOrphans that will disable orphans
+// while the workspace is dragging. This enables the preview to update as soon as
+// a block is dragged, and for a new block to not be enabled until it is attached
```

Comment:
Two nits on this comment:

> This enables the preview to update as soon as a block is dragged,

1. I don't understand the "enables the preview to update as soon as the block is dragged" part... I thought not updating the preview until it was attached to an enabled parent button was the point?
2. Could we modify this to clarify which parts are custom? Something like, "A custom version of Blockly's Events.disableOrphans that retains the functionality of enabling blocks once they are attached to an enabled parent, but modifies a condition so that blocks remain disabled even if they're being dragged"?

Reply:
This change covers two scenarios:
  a. When a block is dragged from the toolbox
  b. When a block is dragged from `when run`
For question 1, this refers to scenario `b`. We weren't updating the block to be `disabled` until it was done being dragged, so the preview update was delayed. We are immediately updating the block to be `disabled` for scenario `a`. The preview is still re-generated, but that code is commented out.

I'll update the comment to be more clear. I only removed one line from the original function, I'll see if I can find a neat way to explain that.

Reply:
Updated the comment, let me know if that clears things up!

Reply:
Yep, thanks, Molly!